### PR TITLE
[feat] World model training using third person games

### PR DESCRIPTION
**Pull request by mignonjia**
## Purpose

Add world-model training helpers for MatrixGame2/Zelda, including LongLive-style streaming long tuning, Zelda validation utilities, action overlays, and synthetic optical-flow validation wiring.

Fixes #

## Changes

- Add `StreamingLongTuningMethod` for LongLive-style streaming long tuning from a self-forcing checkpoint, with long overlapping chunks and streaming context.
- Add MatrixGame2 causal attention sink support for long-context streaming.
- Add Zelda world-model configs, validation metric wiring, and synthetic optical-flow calibration.
- Add validation action overlays and artifact-save handling.
- Document Zelda training/validation data downloads and the suggested `data/zeldam2-clean` local path.

## Test Plan

```bash
# CPU unit lane (mirrors CI collection)
pytest fastvideo/tests/dataset/ fastvideo/tests/train/ \
  --ignore=fastvideo/tests/train/models --ignore=fastvideo/tests/train/methods -q

# Per-method GPU smoke + CPU helper tests (H200, single GPU)
pytest fastvideo/tests/train/methods/test_streaming_long_tuning.py -q

# Lint on all changed files
pre-commit run --files $(git diff origin/main...HEAD --name-only)
```

## Test Results

<details>

- CPU unit lane: **129 passed** (includes new dataset multi-root/invalid-spec tests,
  dict `data_path` override test, trainer validation dispatch test, validation
  callback metric/overlay tests — 27 passed in `callbacks/test_validation.py`)
- `test_streaming_long_tuning.py`: **11 passed** on H200 — GPU smoke walks the
  self-forcing stage, first/overlapped/final streaming chunks, and state reset;
  grad-norm reference seeded for H200 (`test_streaming_long_tuning[H200] = 0.0049`,
  L40S key to be seeded via `modal run ...::seed_grad_norm_references`)
- Peak GPU memory during the smoke test: **27.4 GB** (fits the 48 GB L40S CI runner)
- pre-commit on all 35 changed files: **all hooks passed**
- SSIM: `test_matrixgame2_similarity.py` generation runs end-to-end locally on H200

</details>

## Review Notes

- Kept the independent no-grad critic rollout in `_losses_for_batch`, matching existing `DMD2Method` and `SelfForcingMethod` behavior.
- Left `_dmd_loss_masked` denominator unchanged because it already clamps with `clamp_min(1e-6)`.

## Checklist

- [x] I ran pre-commit on the changed PR files and fixed all issues
- [x] I added or updated tests for my changes
- [x] I updated documentation if needed
- [x] I considered GPU memory impact of my changes

**For model/pipeline changes, also check:**
- [x] I verified SSIM regression tests pass
- [x] I updated the support matrix if adding a new model

### CODE REVIEW

File: fastvideo/train/methods/distribution_matching/streaming_long_tuning.py [hunk 1]
```diff
@@ -0,0 +1,1464 @@
...
+            "condition_image_embeds": None,
+            "generator_chunk_start": int(current),
+            "generator_chunk_end": int(current + new_frames),
+            "generator_action_frame_end": int((current + new_frames - 1) * temporal_compression_ratio + 1),
+            "window_label": f"window_{score_frame_start}_{score_frame_start + int(full_chunk.shape[1])}",
+        }
+        state.previous_latents = full_chunk.detach()[:, -chunk_size:]
+
+        if not dist.is_initialized() or dist.get_rank() == 0:
+            print(
```
Comment (H1yori233) [inline fastvideo/train/methods/distribution_matching/streaming_long_tuning.py]: should use `logger`

File: fastvideo/models/dits/matrixgame2/action_module.py [hunk 2]
```diff
@@ -91,6 +91,27 @@ def _padding_q_k_v(tensor: torch.Tensor, padded_length: int) -> torch.Tensor:
     )
 
 
+def _retain_kv_with_sink(
```
Comment (H1yori233) [inline fastvideo/models/dits/matrixgame2/action_module.py]: There are duplicate functions in `fastvideo/models/dits/matrixgame2/causal_model.py`; they should be merged.

File: fastvideo/train/methods/distribution_matching/streaming_long_tuning.py [hunk 3]
```diff
@@ -0,0 +1,1464 @@
...
+                        min_new,
+                        where=("multi_phased_distill_schedule"
+                               f"[{idx}].streaming_min_new_frame"),
+                    )),
+                    streaming_fixed_overlap_latents=(None if fixed_overlap is None else _as_int(
+                        fixed_overlap,
+                        where=("multi_phased_distill_schedule"
+                               f"[{idx}].streaming_fixed_overlap_latents"),
+                    )),
+                    train_first_chunk=_as_bool(
```
Comment (H1yori233) [inline fastvideo/train/methods/distribution_matching/streaming_long_tuning.py]: `train_first_chunk` is read but never used, so it should be deleted.

File: fastvideo/train/methods/distribution_matching/streaming_long_tuning.py [hunk 4]
```diff
@@ -0,0 +1,1245 @@
...
+            generator_pred_x0 = self._student_rollout(
+                training_batch,
+                with_grad=False,
+            )
+
+        fake_score_loss, critic_ctx, critic_outputs = (self._critic_flow_matching_loss_for_x0(
+            generator_pred_x0,
+            training_batch,
+            chunk_mask=chunk_mask,
+        ))
```
[the remark this answers is not in the document]
Comment (mignonjia) [reply fastvideo/train/methods/distribution_matching/streaming_long_tuning.py:532]: We need one rollout for generator, and another for critic. This matches dmd2 and self forcing.

**Review by alexzms — APPROVED** [review]
